tests/contrib/kafka/test_kafka.py

```diff
@@ -29,14 +34,28 @@ def process_trace(self, trace):
         # Filter out all poll spans that have no received message
         return (
             None
-            if trace[0].name == "kafka.consume" and trace[0].get_tag("kafka.received_message") == "False"
+            if trace[0].name in {"kafka.consume", "kafka.process"}
+            and trace[0].get_tag("kafka.received_message") == "False"
             else trace
         )
 
 
 dd_tracer.configure(settings={"FILTERS": [KafkaConsumerPollFilter()]})
 
 
+@pytest.fixture()
+def kafka_topic(request):
```

Comment:
The kafka tests needed to be changed quite a bit to support running with different subprocesses.  Prior to these changes, the subprocess code would flakily segfault due to race conditions with creating subscriptions (confluent_kafka will segfault on some machines if you try to consume from a non-existent topic).